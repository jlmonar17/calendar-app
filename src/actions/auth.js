import Swal from "sweetalert2";
import { fetchWithoutToken, fetchWithToken } from "../helpers/fetch";
import { types } from "../types/types";

export const startLogin = (email, password) => {
    return async (dispatch) => {
        const response = await fetchWithoutToken(
            "auth",
            { email, password },
            "POST"
        );
        const body = await response.json();

        if (body.ok) {
            localStorage.setItem("token", body.token);
            localStorage.setItem("token-init-date", new Date().getTime());

            const user = {
                uid: body.uid,
                name: body.name,
            };

            dispatch(login(user));
        } else {
            Swal.fire("Error", body.msg, "error");
        }
    };
};

export const startRegister = (email, name, password) => {
    return async (dispatch) => {
        const response = await fetchWithoutToken(
            "auth/new",
            { email, name, password },
            "POST"
        );
        const body = await response.json();

        if (body.ok) {
            localStorage.setItem("token", body.token);
            localStorage.setItem("token-init-date", new Date().getTime());

            const user = {
                uid: body.uid,
                name: body.name,
            };

            dispatch(login(user));
        } else {
            Swal.fire("Error", body.msg, "error");
        }
    };
};

const login = (user) => ({
    type: types.authLogin,
    payload: user,
});

export const startChecking = () => {
    return async (dispatch) => {
        const response = await fetchWithToken("auth/renew");
        const body = await response.json();

        console.log(body);

        if (body.ok) {
            localStorage.setItem("token", body.token);
            localStorage.setItem("token-init-date", new Date().getTime());

            const user = {
                uid: body.uid,
                name: body.name,
            };

            dispatch(login(user));
        } else {
            dispatch(checkingFinish());
        }
    };
};

const checkingFinish = () => ({
    type: types.authCheckingFinish,
});

export const startLogout = () => {
    return (dispatch) => {
        localStorage.clear();

        dispatch(logout());
    };
};

const logout = () => ({
    type: types.authLogout,
});
