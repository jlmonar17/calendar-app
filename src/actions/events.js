import Swal from "sweetalert2";
import { fetchWithToken } from "../helpers/fetch";
import { prepareEvents } from "../helpers/prepareEvents";
import { types } from "../types/types";

export const eventStartAddNew = (event) => {
    return async (dispatch, getState) => {
        const { uid, name } = getState().auth;

        const res = await fetchWithToken("events", event, "POST");
        const body = await res.json();

        if (body.ok) {
            event.id = body.event.id;
            event.user = {
                _id: uid,
                name,
            };

            dispatch(eventAddNew(event));
        }
    };
};

const eventAddNew = (event) => ({
    type: types.eventAddNew,
    payload: event,
});

export const eventSetActive = (event) => ({
    type: types.eventSetActive,
    payload: event,
});

export const eventClearActiveEvent = () => ({
    type: types.eventClearActiveEvent,
});

export const eventStartUpdate = (event) => {
    return async (dispatch) => {
        try {
            const response = await fetchWithToken(
                `events/${event.id}`,
                event,
                "PUT"
            );
            const body = await response.json();

            if (body.ok) {
                dispatch(eventUpdated(event));
            } else {
                Swal.fire("Error", body.msg, "error");
            }
        } catch (error) {
            console.log(error);
        }
    };
};

const eventUpdated = (event) => ({
    type: types.eventUpdated,
    payload: event,
});

export const eventStartDelete = () => {
    return async (dispatch, getState) => {
        const { id } = getState().calendar.activeEvent;

        try {
            const response = await fetchWithToken(`events/${id}`, {}, "Delete");
            const body = await response.json();

            if (body.ok) {
                dispatch(eventDeleted());
            } else {
                Swal.fire("Error", body.msg, "error");
            }
        } catch (error) {
            console.log(error);
        }
    };
};

const eventDeleted = () => ({
    type: types.eventDeleted,
});

export const eventStartLoading = () => {
    return async (dispatch) => {
        try {
            const response = await fetchWithToken("events");
            const body = await response.json();

            const events = prepareEvents(body.events);

            if (body.ok) {
                dispatch(eventsLoaded(events));
            }
        } catch (error) {
            console.log(error);
        }
    };
};

const eventsLoaded = (events) => ({
    type: types.eventsLoaded,
    payload: events,
});

export const eventLogout = () => ({
    type: types.eventLogout,
});
