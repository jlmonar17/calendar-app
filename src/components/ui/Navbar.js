export const Navbar = () => {
    return (
        <div className="navbar navbar-dark bg-dark mb-4">
            <span className="navbar-brand">Jos√©</span>

            <button className="btn btn-outline-danger">
                <li className="fas fa-sign-out-alt "></li>
                <span> Logout</span>
            </button>
        </div>
    );
};
