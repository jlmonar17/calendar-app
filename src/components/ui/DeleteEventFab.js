import { useDispatch, useSelector } from "react-redux";
import { eventDeleted } from "../../actions/events";

export const DeleteEventFab = () => {
    const dispatch = useDispatch();

    const handleDelete = () => {
        dispatch(eventDeleted());
    };

    return (
        <button className="btn btn-danger fab-danger" onClick={handleDelete}>
            <i className="fa fa-trash"></i>
            <span> Delete event</span>
        </button>
    );
};
