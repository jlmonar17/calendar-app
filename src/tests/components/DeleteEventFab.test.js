import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { DeleteEventFab } from "../../components/ui/DeleteEventFab";
import { Provider } from "react-redux";
import { eventStartDelete } from "../../actions/events";
jest.mock("../../actions/events");

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);

store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab />
    </Provider>
);

describe("Tests for <DeleteEventFab/>", () => {
    test("Should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Should call eventStartDelete on click button", () => {
        wrapper.find("button").simulate("click");

        expect(store.dispatch).toHaveBeenCalled();
        expect(eventStartDelete).toHaveBeenCalled();
    });
});
