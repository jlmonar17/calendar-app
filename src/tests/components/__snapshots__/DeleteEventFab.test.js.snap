// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <DeleteEventFab/> Should render correctly 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fa fa-trash"
  />
  <span>
     Delete event
  </span>
</button>
`;
